### PR TITLE
[WIP]fix aggregation type for per process stats and system stats.

### PR DESCRIPTION
**Pull request by rghetia**
Also recording delta for sum aggregation by preserving previous values.

### CODE REVIEW

File: receiver/vmmetricsreceiver/vm_metrics_collector.go [hunk 1]
```diff
@@ -127,7 +136,14 @@ func (vmc *VMMetricsCollector) scrape() {
 			mSystemCPUSeconds.M(cpuStat.System),
 			mIdleCPUSeconds.M(cpuStat.Idle),
 			mIowaitCPUSeconds.M(cpuStat.Iowait))
+		if prevStat != nil {
+			stats.Record(ctx, mProcessesCreated.M(int64(stat.ProcessCreated - prevStat.ProcessCreated)))
```
Comment (songy23) [inline receiver/vmmetricsreceiver/vm_metrics_collector.go]: Should `ProcessRunning` and `ProcessBlocked` be updated too?

**Comment by songy23** [comment]
Also please format the files.

File: receiver/vmmetricsreceiver/vm_metrics_collector.go (hunk 1 above)
Comment (rghetia) [reply receiver/vmmetricsreceiver/vm_metrics_collector.go]: No. I checked, they are instantaneous values.

**Comment by rghetia** [comment]
@songy23 , I reverted back to aggregation sum for system (user, system, idle, nice, iowait). But added the delta recording for each.

VM stat is correctly reported now. However, when I run the agent as daemon set and mount /proc from host as /data/proc on agent container, vm stats are correct but agent process stat is not. If I don't mount /data/proc, process stat is correct but vm stat is not.

**Comment by rghetia** [comment]
@songy23, @pjanotti thx for the review. There is one more update which contains following two changes.
1. 'prev*' was not correctly initialized in earlier commit. It should be set at every poll. 
2. The process level cpu_seconds has only fractional seconds between two polls. Converting fractions to int64 makes it zero. Hence changed it to float64.

PTAL.

**Comment by rghetia** [comment]
@songy23 or @pjanotti can you please merge it? I don't have access.

**Pull request merged**